src/objects.rs

```diff
@@ -231,3 +234,156 @@ where
 
     Ok(())
 }
+
+#[derive(Serialize, Deserialize, Debug)]
+// This serde macro adds the "operation" field to each variant's structure when
+// it is serialized.
+#[serde(tag = "operation")]
+#[serde(rename_all = "camelCase")]
+pub enum BatchRequest {
+    Put(BatchPutOp),
+    Update(BatchUpdateOp),
+    Delete(BatchDeleteOp),
+    DeleteMany(BatchDeleteManyOp),
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct BatchPutOp {
+    pub bucket: String,
+    pub options: MethodOptions,
+    pub key: String,
+    pub value: Value,
+}
+
+/// For now we only support Put operations
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct BatchUpdateOp {
```

Comment:
I added the framework for these other operations, and the ones I've tested work, but I haven't exhaustively tested them yet, so I've blocked their use on line 293 below.  If we get some help or someone needs them everything is here, they just need tests/examples.